script.js

```diff
@@ -1,4 +1,285 @@
+var makeDeck = function () {
+  var cardDeck = [];
+  var suits = ["hearts", "diamonds", "clubs", "spades"];
+  var suitsEmoji = ["♥︎", "♦︎", "♣︎", "♠︎"];
+  var suitIndex = 0;
+  while (suitIndex < suits.length) {
+    var currentSuit = suits[suitIndex];
+    var currentEmoji = suitsEmoji[suitIndex];
+    var rankCounter = 1;
+    while (rankCounter <= 13) {
+      var cardName = rankCounter;
+      if (cardName == 1) {
+        cardName = "ace";
+      } else if (cardName == 11) {
+        cardName = "jack";
+      } else if (cardName == 12) {
+        cardName = "queen";
+      } else if (cardName == 13) {
+        cardName = "king";
+      }
+
+      var currentValue = rankCounter;
+      if (currentValue == 11 || currentValue == 12 || currentValue == 13) {
+        currentValue = 10;
+      }
+
+      var card = {
+        name: cardName,
+        suit: currentSuit,
+        rank: rankCounter,
+        emoji: currentEmoji,
+        blackjackValue: currentValue,
+      };
+      cardDeck.push(card);
+      rankCounter += 1;
+    }
+    suitIndex += 1;
+  }
+  return cardDeck;
+};
+
+var getRandomIndex = function (max) {
+  return Math.floor(Math.random() * max);
+};
+
+var shuffleCards = function (cardDeck) {
+  var currentIndex = 0;
+  while (currentIndex < cardDeck.length) {
+    var randomIndex = getRandomIndex(cardDeck.length);
+    var randomCard = cardDeck[randomIndex];
+    var currentCard = cardDeck[currentIndex];
+    cardDeck[currentIndex] = randomCard;
+    cardDeck[randomIndex] = currentCard;
+    currentIndex = currentIndex + 1;
+  }
+  return cardDeck;
+};
+
+var reset = function () {
+  newDeck = shuffleCards(makeDeck());
+  allCardsDealt = [];
+  playersBlackjackValue = [];
+  playerTurn = 1;
+  gameMode = "draw cards";
+};
+
+var newDeck = shuffleCards(makeDeck());
+var numOfPlayers = 2; // Inclusive of dealer
+var dealer = numOfPlayers; // Dealer is always the last to play
+var allCardsDealt = []; // Holds array of card object arrays
+var playersBlackjackValue = []; // Holds array of final numbers
+var playerTurn = 1; // Default start with player 1
```

Comment:
setting playerTurn to start at 0 can be make it easier for accessing array inputs, just got to ${playerTurn+1} when outputting messages